modules/data-entry/src/main/frontend/src/vocabQuery/InfoBrowser.jsx

```diff
@@ -0,0 +1,205 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+import classNames from "classnames";
+import React, { useRef, useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { Snackbar, SnackbarContent } from "@material-ui/core";
+import { withStyles } from "@material-ui/core";
+
+import VocabularyBrowser from "./browse.jsx";
+import InfoBox from "./infoBox.jsx";
+import { REST_URL, MakeRequest } from "./util.jsx";
+import QueryStyle from "./queryStyle.jsx";
+
+// Component that renders a vocabulary info box and browser.
+//
+function InfoBrowser(props) {
+  const { browserOpen, onClose, infoPath, infoButtonRefs, infoboxRef, browserRef, classes } = props;
+
+  const [termInfoVisible, setTermInfoVisible] = useState(false);
+  const [term, setTerm] = useState({});
+
+  const [browseID, setBrowseID] = useState("");
+  const [browsePath, setBrowsePath] = useState("");
+  const [browserOpened, setBrowserOpened] = useState(!!browserOpen);
+
+  const [vocab, setVocab] = useState({});
+
+  const [snackbarVisible, setSnackbarVisible] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState('');
+
+  // Strings used by the info box
+  const [infoAboveBackground, setInfoAboveBackground] = useState(false);
+  const [infoAnchor, setInfoAnchor] = useState(null);
```

Comment:
These are not strings. `Info box properties`?
```suggestion
  // Info box properties
  const [infoAboveBackground, setInfoAboveBackground] = useState(false);
  const [infoAnchor, setInfoAnchor] = useState(null);
```